public/components/integrations/components/added_integration_table.tsx

```diff
@@ -100,36 +84,38 @@ export function AddedIntegrationsTable(props: AddedIntegrationsTableProps) {
     });
   }
 
-  async function deleteAddedIntegration(integrationInstance: string, name: string) {
-    http
-      .delete(`${INTEGRATIONS_BASE}/store/${integrationInstance}`)
-      .then(() => {
-        setToast(`${name} integration successfully deleted!`, 'success');
-        props.setData({
-          hits: props.data.hits.filter((i) => i.id !== integrationInstance),
-        });
-      })
-      .catch((_err) => {
-        setToast(`Error deleting ${name} or its assets`, 'danger');
-      })
-      .finally(() => {
-        window.location.hash = '#/installed';
-      });
+  async function deleteAddedIntegrations(selectedItems: any[]) {
+    const deletePromises = selectedItems.map(async (item) => {
+      try {
+        await http.delete(`${INTEGRATIONS_BASE}/store/${item.id}`);
+        setToast(`${item.name} integration successfully deleted!`, 'success');
+      } catch (error) {
+        setToast(`Error deleting ${item.name} or its assets`, 'danger');
+      }
+    });
+
+    await Promise.all(deletePromises);
```

Comment:
do we need this line? I see an await on line number 90

Reply:
Removed